sky/design_docs/cluster_name.md

```diff
@@ -11,16 +11,19 @@ when the following situation happens, it can be annoying:
 
 After #2403, SkyPilot will automatically truncate cluster names to comply with the
 naming length limit of the cloud, and append a user hash to the end of the name to
-distinguish clusters from different users.
+distinguish clusters from different users (we make sure the final result after the append is within length limit).
 
 ## Details
 
 Cluster names used in different situations:
 - CLI/API/UI: the name users assign to the cluster, i.e., the original cluster name
-    without any modification.
+    without any modification. `cluster_name` in code.
 - Lower-level components: for the components that directly interact with the cloud
     provider, the newly generated name will be used, including, the underlying
-    `ray up`, `provision` module, etc.
+    `ray up`, `provision` module, etc. `cluster_name_on_cloud` in code.
+- Cluster name on cloud: names of clusters on the cloud will always be
+    `cluster_name_on_cloud`, including the cluster name shown in cloud consoles,
+    the cluster name in the tag (ray-cluster-name), etc.
```

Comment:
I'd suggest the following 2 items rather than 3 concepts:

--

There are two cluster name concepts:
- **SkyPilot cluster name**: This is the original "display" name users pick (or generated by the system) for each SkyPilot cluster. This is used in our CLI/API (e.g., `sky launch -c <cluster name>`). 
  - `cluster_name` in code. Used in all user-facing logging.
- **Cluster name on cloud**: This is the auto-generated name passed to the cloud. We auto-process the display name above to produce this new name to ensure (1) distinguishing across users, in case they choose the same cluster name; (2) the cluster name on cloud is within length limit.
  - `cluster_name_on_cloud` in code. Used in all modules that communicated with the cloud provider, e.g., `ray up`, node providers/`provision_lib` module, etc.
  - Recorded in the VM tag `ray-cluster-name`. (Note that the VM names include it as a substring, but may have additional prefixes/suffixes.)

In most cases, users need only be aware of the former.

Reply:
Good point! Updated. Thanks!